### PR TITLE
metabase: Support new match types

### PR DESCRIPTION
**Pull request by cthulhu-rider**
- [x] support of `STRING_NOT_EQUAL` in metabase
- [x] support of `NOT_PRESENT` in metabase
- [x] define and implement format in CLI (#362)

### CODE REVIEW

File: pkg/local_object_storage/metabase/select.go [hunk 1]
```diff
@@ -223,10 +212,57 @@ func (db *DB) selectFastFilter(
...
+	for _, fns := range mBucketNaming {
+		for _, fn := range fns {
+			names = append(names, fn(cid))
+		}
+	}
+
+	return
+}
+
+func bucketNamesForType(cid *container.ID, mType object.SearchMatchType, typeVal string) (names [][]byte) {
```
Comment (alexvanin) [inline pkg/local_object_storage/metabase/select.go:241]: Maybe `allBucketNames` can be reused in this function if we pass `object.MatchNotPresent` type. Just a thought.

File: pkg/local_object_storage/metabase/select.go [hunk 2]
```diff
@@ -271,6 +307,56 @@ func (db *DB) selectFromFKBT(
...
+	fNum int, // index of filter
+) {
+	exclBktRoot := tx.Bucket(name)
+
+	mExcl := make(map[string]struct{})
+
+	bktExcl := tx.Bucket(name)
+	if bktExcl != nil {
+		_ = bktExcl.ForEach(func(k, _ []byte) error {
+			exclBktLeaf := exclBktRoot.Bucket(k)
```
Comment (alexvanin) [inline pkg/local_object_storage/metabase/select.go:328]: Can't you do `exclBktLeaf := bktExcl.Bucket(k)` there?

**Review by alexvanin — APPROVED** [review]

File: pkg/local_object_storage/metabase/select.go (hunk 2 above)
Comment (cthulhu-rider) [reply pkg/local_object_storage/metabase/select.go]: Yes, fixed

File: pkg/local_object_storage/metabase/select.go (hunk 1 above)
Comment (cthulhu-rider) [reply pkg/local_object_storage/metabase/select.go:241]: I thought about that, but I decided that `allBucketNames` could have simpler logic, which is necessary when processing `MatchNotPresent`. Call `bucketNamesForType(cid, object.MatchNotPresent, "")` will not give the desired result.

**Pull request merged**